### PR TITLE
Compiler fixes

### PR DESCRIPTION
**Pull request by DrChat**
* Fix double-swapping in OPCODE_LOAD
* Add code to enforce page protections (but commented because it introduces a load of constants)

### CODE REVIEW

File: src/xenia/cpu/hir/value.cc [hunk 1]
```diff
@@ -77,8 +77,47 @@ uint64_t Value::AsUint64() {
...
 
 void Value::Cast(TypeName target_type) {
-  // TODO(benvanik): big matrix.
-  assert_always();
+  switch (type) {
+    case INT32_TYPE:
+      switch (target_type) {
+        case FLOAT32_TYPE:
+          // ???
+          constant.f32 = constant.f32;
```
Comment (benvanik) [inline src/xenia/cpu/hir/value.cc]: No question marks needed - or the assignment - as a constant op this is fine just being the type change

File: src/xenia/cpu/xex_module.cc [hunk 2]
```diff
@@ -281,6 +281,34 @@ bool XexModule::Load(const std::string& name, const std::string& path,
     }
   }
 
+  // Setup memory protection.
+  // TODO: This introduces a load of constants into the JIT, and Xenia isn't
```
Comment (benvanik) [inline src/xenia/cpu/xex_module.cc]: What other things are missing? this is awesome!

File: src/xenia/cpu/compiler/passes/constant_propagation_pass.cc [hunk 3]
```diff
@@ -153,6 +153,14 @@ bool ConstantPropagationPass::Run(HIRBuilder* builder) {
             i->Remove();
           }
           break;
+        case OPCODE_CONVERT:
+          if (i->src1.value->IsConstant()) {
+            TypeName target_type = v->type;
+            v->set_from(i->src1.value);
+            v->Convert(target_type, ROUND_TO_NEAREST);
```
Comment (benvanik) [inline src/xenia/cpu/compiler/passes/constant_propagation_pass.cc]: Pass the rounding mode in i->flags

File: src/xenia/cpu/xex_module.cc (hunk 2 above)
Comment (DrChat) [inline src/xenia/cpu/xex_module.cc]: Over half of x64_sequences.cc isn't setup to handle constants. We probably need a rewrite or something.

Comment (benvanik) [inline src/xenia/cpu/xex_module.cc]: Ah yeah - maybe I can help with some of that. I still want to rework that file to make it more readable/etc. Adding a proper LIR would be nice, too.

**Pull request merged**